test/lib/foreman/logging_test.rb

```diff
@@ -7,6 +7,17 @@ def test_configure_once
     end
   end
 
+  def test_environment_not_found_yields_defaults
+    assert_nothing_raised do
+      Foreman::Logging.instance_variable_set("@configured", false)
+      Foreman::Logging.configure(:environment => 'foo')
+    end
+    default_config = YAML.load_file('config/logging.yaml')[:default]
+    # The only difference between the default options and 'foo's is the log filename
+    filename = Foreman::Logging.instance_variable_get("@config").to_a - default_config.to_a
```

Comment:
I'm not sure I follow why you do this, could you just pass the full hash into assert_equal?